File: leetcode/python/0-99/62_unique_path.py
import sys
from typing import List


class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0 for j in range(n)] for i in range(m)]

        for i in range(m):
            dp[i][0] = 1
        for j in range(n):
            dp[0][j] = 1

        for i in range(1, m):
            for j in range(1, n):
                dp[i][j] = dp[i - 1][j] + dp[i][j - 1]

        return dp[m - 1][n - 1]

solution = Solution()
print(solution.uniquePaths(m = 3, n = 7))

File: leetcode/c++/000-099/62_unique_path.cpp
#include <string>
#include <iostream>
#include <stack>
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int uniquePaths(int m, int n) {
        vector dp(m, vector(n, 1));

        for (int row = 1; row < m; ++row) {
            for (int col = 1; col < n; ++col) {
                dp[row][col] = min(dp[row - 1][col] + 1, dp[row][col - 1]);
            }
        }

        return dp[m - 1][n - 1];
    }
};

int main() {
    auto s = Solution();

    for (const vector<string> st = {"eat", "tea", "tan", "ate", "nat", "bat"}; auto &e: s.groupAnagrams(st)) {
        for (auto &x: e) cout << x << ' ';
        cout << '\n';
    }

    for (const vector<string> st2 = {""}; auto &e: s.groupAnagrams(st2)) {
        for (auto &x: e) cout << x << ' ';
        cout << '\n';
    }

    for (const vector<string> st3 = {""}; auto &e: s.groupAnagrams(st3)) {
        for (auto &x: e) cout << x << ' ';
        cout << '\n';
    }
}
